packages/apollo-server-core/src/__tests__/ApolloServerBase.test.ts

```diff
@@ -30,6 +30,53 @@ describe('ApolloServerBase construction', () => {
     ).not.toThrow();
   });
 
+  it('succeeds when passed a graphVariant in construction', () => {
+    let serverBase;
+    expect(
+      () =>
+        serverBase = new ApolloServerBase({
+          schema: buildServiceDefinition([{ typeDefs, resolvers }]).schema,
+          engine: {
+            graphVariant: 'foo',
+            apiKey: 'not:real:key',
+          },
+        }),
+    ).not.toThrow();
+    (serverBase as unknown as ApolloServerBase).stop();
+  });
+
+  it('spits out a deprecation warning when passed a schemaTag in construction', () => {
+    const spyConsoleWarn = jest.spyOn(console, 'warn').mockImplementation();
+    let serverBase;
+    expect(
+      () =>
+        serverBase = new ApolloServerBase({
+          schema: buildServiceDefinition([{ typeDefs, resolvers }]).schema,
+          engine: {
+            schemaTag: 'foo',
+            apiKey: 'not:real:key',
+          },
+        }),
+    ).not.toThrow();
+    expect(spyConsoleWarn).toBeCalled();
+    spyConsoleWarn.mockRestore();
+    (serverBase as unknown as ApolloServerBase).stop();
```

Comment:
Same as above suggestion, hopefully one-click suggestion win:

```suggestion
    expect(
      () =>
        new ApolloServerBase({
          schema: buildServiceDefinition([{ typeDefs, resolvers }]).schema,
          engine: {
            schemaTag: 'foo',
            apiKey: 'not:real:key',
          },
        }).stop()
    ).not.toThrow();
    expect(spyConsoleWarn).toBeCalled();
    spyConsoleWarn.mockRestore();
```